Remotion/Web/Core/Res/Themes/NovaViso/HTML/TreeView.css

```diff
@@ -1,570 +1,588 @@
-ul.treeViewTopLevelNodeChildren
+ul.treeViewRoot a
 {
-  margin: 0;
-  padding: 0;
-  list-style: none;
+  color: black;
+  text-decoration: none;
 }
 
-/* Applied to the node's children area. */
-ul.treeViewNodeChildren
+ul.treeViewRoot .treeViewNode
 {
-  background-image: url(../Image/sprite.svg#TreeViewI);
-  background-size: 16px 16px;
-  background-repeat: repeat-y;
-  margin: 0;
-  padding: 0;
-  list-style: none;
+  position: relative;
 }
 
-/* Applied to the node's children area if the nodes should not be connected by lines. */
-ul.treeViewNodeChildrenNoLines
+ul.treeViewRoot,
+ul.treeViewRoot ul.treeViewNodeChildren,
+ul.treeViewRoot ul.treeViewNodeChildrenNoLines,
+ul.treeViewRoot ul.treeViewTopLevelNodeChildren
 {
-  margin: 0;
   padding: 0;
   list-style: none;
 }
 
-ul.treeViewRoot li
+ul.treeViewRoot span.treeViewNodeHeadSelected a:focus,
+ul.treeViewRoot li[role=treeitem]
 {
   outline: none;
 }
 
-ul.treeViewRoot > li,
-ul.treeViewTopLevelNodeChildren > li
+ul.treeViewRoot li:focus > span.treeViewNode span.treeViewNodeHead a > span,
+ul.treeViewRoot li:focus > span.treeViewNode span.treeViewNodeHead a:hover > span
 {
+  text-decoration: underline;
 }
 
-ul.treeViewNodeChildren > li,
-ul.treeViewNodeChildrenNoLines > li
+ul.treeViewRoot span.treeViewNode:hover .treeViewNodeHead:before,
+ul.treeViewRoot span.treeViewNode .treeViewNodeHeadSelected:before,
+ul.treeViewRoot li:focus > span.treeViewNode span.treeViewNodeHead:before,
+ul.treeViewRoot li:focus > span.treeViewNode span.treeViewNodeHeadSelected:before
 {
+  background-color: var(--color-highlight);
 }
 
-/* Applied to the node itself */
-span.treeViewNode
+ul.treeViewRoot .treeViewNodeHead:before,
+ul.treeViewRoot .treeViewNodeHeadSelected:before
 {
-  cursor: default;
-  display: inline-block;
-  width: 100%;
-  box-sizing: border-box;
+  content: " ";
+  position: absolute;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+  z-index: -1;
+}
+
+ul.treeViewRoot .treeViewNode,
+ul.treeViewRoot .treeViewNodeHead > a,
+ul.treeViewRoot .treeViewNodeHeadSelected > a
+{
+  display: flex;
+  align-items: center;
+  padding-top: calc(0.5 * var(--remotion-themed-padding));
+  padding-bottom: calc(0.5 * var(--remotion-themed-padding));
+}
+
+ul.treeViewRoot span.treeViewNode > a img.Icon
+{
+  height: var(--treeview-chevron-size);
+  width: var(--treeview-chevron-size);
+  margin-right: var(--treeview-chevron-gap);
 }
 
-/* We want the tree node to span the entire line to better handle clicking and styling of the selected line.
-  Since it is not possible to use a nesting-count in CSS, we need to do an explicit nesting for a reasonable depth (50 levels). 
-  Once the maximum depth is reached, the formatting and behavior will revert to nesting via parent padding. */
+/*region Padding calculation*/
 ul.treeViewRoot span.treeViewNode
 {
-  padding-left: calc(19px * 0);
+  padding-left: calc(var(--treeview-padding)  * 0);
```

Comment:
I just had an idea, could we render the nesting level as a css variable on the respective node:
https://stackoverflow.com/questions/20490704/combine-calc-with-attr-in-css

That way, we solve this mess once and for all. When it works, we should also update the NovaGray theme.